GuessTheNumber.java

```diff
@@ -21,4 +24,129 @@
  * Before the next game, the top 3 scores (which are greater than 0) will be printed
  */
 public class GuessTheNumber {
+
+    private static byte move = 0;
+    private static byte samePosition;
+
+    public static void main(String[] args) {
+
+        Scanner input = new Scanner(System.in);
+        byte[] baseArray = new byte[4];
+        byte[] guessArray = new byte[4];
+        int guessedNumber;
+
+        System.out.printf("Guess the number: start a new game? Yes / No : ");
+        String confirm = input.next();
+
+        if (!confirm.matches("Yes") && (!confirm.matches("yes")) && (!confirm.matches("y"))) {
+            return;
+        }
+
+        System.out.println("A 4-digit number has been chosen. Only digits 0-9 are allowed without duplicates.\n" +
+                "This number is guessed by you via multiple attempts. An attempt consists in a guessed number \n" +
+                "and the computer's reply. The computer must tell you how many digits have you guessed\n" +
+                "on the same position, and how many digits have you guessed on a different position. \n" +
+                "using that reply you must guess the number using as few moves as possible. Good luck! \n");
+
+        baseArray = populateBaseArray(baseArray);
+
+        while (samePosition != 4) {
+            guessedNumber = getUserGuess(input);
+            guessArray = populateGuessArray(guessArray, guessedNumber);
+            samePosition = tryUserGuess(baseArray,guessArray);
+        }
+        System.out.println("You've guessed the number, Good Job!");
+    }
+
+    private static byte[] populateBaseArray(byte[] baseArray) {
+
+        Random randomNum = new Random();
+        byte[] numbersUsed = new byte[10];
+        int i = 0;
+        int nextInt;
+
+        System.out.printf("Computer's num: (");
+
+        while (i < baseArray.length) {
+            nextInt = randomNum.nextInt(10);
+            if (nextInt != 0 && numbersUsed[nextInt] == 0) {

```

Comment:
This way, your number will hold no zeros at all (not only as first position)

Reply:
@ameedf I initially declared it as boolean but then had a type mismatch when i tried to use numbersUsed[nextInt]. Any idea why?